plugin-api/src/main/java/org/sonar/api/issue/Issue.java

```diff
@@ -161,15 +205,25 @@ public interface Issue extends Serializable {
 
   /**
    * See constant values in {@link Issue}.
+   * @deprecated since 10.4 in favor of {@link IssueStatus}
    */
+  @Deprecated(since = "10.4")
   String status();
 
   /**
    * The type of resolution, or null if the issue is not resolved. See constant values in {@link Issue}.
+   * @deprecated since 10.4 in favor of {@link IssueStatus}
    */
   @CheckForNull
+  @Deprecated(since = "10.4")
   String resolution();
 
+  /**
+   * Available list of status can be found in {@link IssueStatus#values()}
+   * @since 10.4
+   */
+  IssueStatus issusStatus();
```

Comment:
We don't need to add a replacement for status in this Class, as it is not used in the plugin-api (like for the impacts)